packages/react-magma-dom/src/components/Select/MultiSelect.tsx

```diff
@@ -45,14 +50,53 @@ export function MultiSelect<T>(props: MultiSelectProps<T>) {
     setFloating,
     setReference,
     isClearable,
+    initialHighlightedIndex,
   } = props;
 
+  function isItemDisabled(item) {
+    return instanceOfItemWithOptionalDisabled(item) && item.disabled;
+  }
+
   function checkSelectedItemValidity(itemToCheck: T) {
     return (
+      !isItemDisabled(itemToCheck) &&
       items.findIndex(i => itemToString(i) === itemToString(itemToCheck)) !== -1
     );
   }
 
+  function isDisabledItemIndex(index) {
```

Comment:
Some of these functions are repeated across MultiSelect and Select-- is it worth having them in a shared file to avoid duplication?

Reply:
There is no need now, using new logic for it